### PR TITLE
feat: use updatecli to update dependencies

### PR DESCRIPTION
**Pull request by amannocci**
## What is the change being made?

- Implement updatecli to update dependencies.

## Why is the change being made?

- Dependabot doesn't support yet uv tooling.

### CODE REVIEW

File: .github/assets/updatecli.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+name: Update uv lock dependencies
```
Comment (v1v) [inline .github/assets/updatecli.yml:1]: folder name `updatecli/updatecli.d`, that is the default name for pipelines and `updatecli/values.d` for config things. See https://github.com/updatecli/updatecli/tree/main/updatecli

File: .github/workflows/updatecli.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+      - uses: elastic/oblt-actions/updatecli/run@v1
+        with:
+          command: --experimental compose diff
+        env:
+          GITHUB_TOKEN: ${{ steps.get_token.outputs.token }}
+
+      - uses: elastic/oblt-actions/updatecli/run@v1
+        with:
+          command: --experimental compose apply
```
Comment (v1v) [inline .github/workflows/updatecli.yml:46]: let's remove experimental, I think we are good without it

File: updatecli-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Config file for `updatecli compose ...`.
+# https://www.updatecli.io/docs/core/compose/
+policies:
+  - name: Local Updatecli Policies
+    config:
+      - .github/updatecli.yml
+    values:
+      - .github/values.d/scm.yml
```
Comment (v1v) [inline updatecli-compose.yaml:8]: I prefer we use `updatecli/updatecli.d/` for config and `updatecli/values.d/scm.yml` for values.

File: .github/workflows/updatecli.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+jobs:
+  compose:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: read
+    steps:
+      - uses: actions/checkout@v4
+
```
Comment (v1v) [inline .github/workflows/updatecli.yml:19]: do you need to configure `uv` so the updatecli pipeline has the right environment where to run the things on? 

```suggestion
      - name: Install environment
        uses: ./.github/workflows/env-install
```

Comment (amannocci) [reply .github/workflows/updatecli.yml:19]: Yes, it will need to be at least installed.

File: .github/updatecli/updatecli.d/updatecli.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    default:
+        kind: github
+        spec:
+            owner: '{{ .scm.owner }}'
+            repository: '{{ .scm.repository }}'
+            branch: '{{ .scm.branch }}'
+            username: '{{ requiredEnv "GITHUB_ACTOR" }}'
+            token: '{{ requiredEnv "GITHUB_TOKEN" }}'
+            commitusingapi: true
+        disabled: false
```
Comment (v1v) [inline .github/updatecli/updatecli.d/updatecli.yml:37]: ident 2 instead 4?

File: .github/updatecli/updatecli.d/updatecli.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+actions:
+    default:
```
Comment (v1v) [inline .github/updatecli/updatecli.d/updatecli.yml:40]: ident 2 instead 4?

**Review by v1v — COMMENTED** [review]
I don't know whether you want to use a fixed version for the updatecli if so you can use the `.tool-versions` file and add the updatecli policy to update it automatically

**Comment by amannocci** [comment]
> I don't know whether you want to use a fixed version for the updatecli if so you can use the `.tool-versions` file and add the updatecli policy to update it automatically

Using the latest updatecli version available shouldn't be a problem.

**Review by v1v — APPROVED** [review]

**Pull request merged**